### DISCUSSION TITLE
3D geological (voxel) Plot

### DISCUSSION THREAD

**Discussion by bhawesh-bks** [pyvista/pyvista#3225] [Q&A] [answered]

**Comment by banesullivan** [comment, marked as answer]
Plotting these data can be straightforward with PyVista `PolyData`. Unfortunately, the data you provided are a seemingly sparse point cloud rather than a lattic of points making a 3D volume like what is shown above. 

You can plot the data you shared as follows:

```py
import pandas as pd
import pyvista as pv
import numpy as np

train = pd.read_csv('train.csv')
target = pd.read_csv('target.csv')

# Get the coordinates
coords = train[['x', 'y', 'z']].values

# Construct PyVista object
pts = pv.PolyData(coords)
# Add field data
for k in [_ for _ in train.keys() if _ not in ['x', 'y', 'z']]:
    pts[k] = train[k].values
pts['target'] = target['target'].values

# Plot in 3D
p = pv.Plotter(notebook=False)
p.add_mesh(pts, scalars='target')
p.show_grid()
p.show()
```
![Screen Shot 2022-08-29 at 9 01 21 AM](https://user-images.githubusercontent.com/22067021/187232419-048cc28d-4fd1-4452-9871-30be7166abab.png)

**Reply by bhawesh-bks** [reply]
Thanks a lot @banesullivan. You are just awesome. It helped me a lot.
